### PR TITLE
feat: add tax CRUD

### PR DESCRIPTION
**Pull request by TanyaLagodich**
## Description

Added tax CRUD on backend

## Related Issues

Issue: https://github.com/idurar/erp-crm/issues/544

## Steps to Test
I have tested it on Postman
Can create new tax with TaxName (string) and TaxValue (number).
Also can provide some other fields such as `isDefault`, `created`, `removed` and `enabled`

So on Postman (or other app) we can:
- create new tax with route POST `/api/taxes/list` and with data `{"taxName": "taxName", "taxValue": 123}`
- then get all list of taxes GET `/api/taxes/list` and being sure that our new tax in the list
- check reading by ID GET `/api/taxes/read/:id`
- check updating the tax PATCH `api/taxes/update/:id`

Provide steps on how to test the changes introduced in this pull request.

## Screenshots (if applicable)

create a tax
Снимок экрана 2023-09-24 в 19 20 17

Read the tax
Снимок экрана 2023-09-24 в 19 20 30

Update the tax
Снимок экрана 2023-09-24 в 19 20 47

Also, this some logic about default tax and enabling it

### Case 1:
We have default tax and we create another one (taxName3) and want to do it default. We update all other tax and make it non default. Default now only new tax

default tax "TaxName2"
Снимок экрана 2023-09-25 в 10 28 30

create new default tax "TaxName3" and "taxName2" has become non default 
Снимок экрана 2023-09-25 в 10 29 24

### Case 2:
if update is isDefault:false , we should update first to be isDefault:true
so we have "taxName3" is default and update it to non default

updated
Снимок экрана 2023-09-25 в 10 30 14

the first document has become default 
Снимок экрана 2023-09-25 в 10 30 01

### Case 3:
if the update is enabled:false and is isDefault:true , we should update first to be isDefault:true
updated tax3
Снимок экрана 2023-09-25 в 10 31 44

the first document has become isDefault Снимок экрана 2023-09-25 в 10 32 38

### case 4:
if the update is enabled:false and it's only one exist , we can't accept to disable
Снимок экрана 2023-09-25 в 10 42 03

## Checklist

- [X] I have tested these changes
- [ ] I have updated the relevant documentation
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the codebase
- [X] My changes generate no new warnings or errors
- [X] The title of my pull request is clear and descriptive

### CODE REVIEW

**Review by salahlalami — CHANGES_REQUESTED** [review]

File: backend/controllers/appControllers/taxController.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+const createCRUDController = require('@/controllers/middlewaresControllers/createCRUDController');
+module.exports = createCRUDController('Tax');
```
[the remark this answers is not in the document]
Comment (TanyaLagodich) [reply backend/controllers/appControllers/taxController.js]: Thanks for your feedback!

I've done it. Please, check it

Also I am thinking may be we should add some validation? As I understand, client will always send full json of tax? I mean with enabled field, right? So if it won't, there is might be some issues

And also maybe when `if the new or update is isDefault:true , we should update all other to be isDefault:false` we should check enabled field? in order to do not just the first tax, but the first and enabled tax?

If it's okey, I will do it quickly

File: backend/controllers/appControllers/taxController.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+delete methods['delete'];
+
+methods.create = async (req, res) => {
+    try {
+        const { isDefault, ...rest } = req.body;
+
+        if (isDefault) {
+            await Model.updateMany({}, { isDefault: false });
+        }
+        const result = await new Model({ isDefault, ...rest }).save();
```
Comment (salahlalami) [inline backend/controllers/appControllers/taxController.js:15]: use same instruction as crudmiddleware :     const result = await new Model(req.body).save();

File: backend/controllers/appControllers/taxController.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (salahlalami) [reply backend/controllers/appControllers/taxController.js]: @TanyaLagodich yes , sure .
just make test to see if it work

**Review by salahlalami — CHANGES_REQUESTED** [review]

File: backend/controllers/appControllers/taxController.js (hunk 2 above)
Comment (TanyaLagodich) [reply backend/controllers/appControllers/taxController.js]: Done!

File: backend/controllers/appControllers/taxController.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (TanyaLagodich) [reply backend/controllers/appControllers/taxController.js]: Ok! I've done it. Tests with screenshots in description of PR

**Review by salahlalami — APPROVED** [review]
@TanyaLagodich Thank you

**Pull request merged**